File: IDATT2106 Systemutvikling 2 med smidig prosjekt/backend/src/main/java/com/ntnu/gidd/security/validator/JwtTokenValidator.java
package com.ntnu.gidd.security.validator;


import com.ntnu.gidd.exception.InvalidJwtToken;
import com.ntnu.gidd.model.RefreshToken;
import com.ntnu.gidd.service.token.RefreshTokenService;
import lombok.AllArgsConstructor;
import org.springframework.stereotype.Component;

@Component
@AllArgsConstructor
public class JwtTokenValidator implements TokenValidator {

    private RefreshTokenService refreshTokenService;

    @Override
    public void validate(String jti) {
        RefreshToken refreshToken = refreshTokenService.getByJti(jti);
        if (!refreshToken.isValid())
            throw new InvalidJwtToken();
    }
}

File: IDATT2105 Full-stack applikasjonsutvikling/Rombestilling/backend/src/main/kotlin/ntnu/idatt2105/security/validator/JwtTokenValidator.kt
package ntnu.idatt2105.security.validator

import ntnu.idatt2105.security.exception.InvalidJwtToken
import ntnu.idatt2105.security.service.RefreshTokenService
import ntnu.idatt2105.security.token.RefreshToken
import org.springframework.stereotype.Component
@Component
class JwtTokenValidator(val refreshTokenService: RefreshTokenService) : TokenValidator {

    override fun validate(jti: String) {
        val refreshToken: RefreshToken = refreshTokenService.getByJti(jti)
        if (!refreshToken.isValid) throw InvalidJwtToken()
    }
}
